python/cudf/cudf/utils/cudautils.py

```diff
@@ -551,21 +551,55 @@ def gpu_round(in_col, out_col, decimal):
     round_val = 10 ** (-1.0 * decimal)
 
     if i < in_col.size:
-        if not in_col[i]:
+        current = in_col[i]
+
+        newval = current // round_val * round_val
+        remainder = fmod(current, round_val)
+
+        if remainder != 0 and remainder > (0.5 * round_val) and current > 0:
+            newval = newval + round_val
+            out_col[i] = newval
+
+        elif (
+            remainder != 0
+            and abs(remainder) < (0.5 * round_val)
+            and current < 0
+        ):
+            newval = newval + round_val
+            out_col[i] = newval
+
+        else:
+            out_col[i] = newval
+
+
+@cuda.jit
+def gpu_round_masked(in_col, out_col, mask, decimal):
+    i = cuda.grid(1)
+    round_val = 10 ** (-1.0 * decimal)
+
+    if i < in_col.size:
+        valid = mask_get(mask, i)
+        current = in_col[i]
+
+        if not valid:
             out_col[i] = np.nan
```

Comment:
Why are we having the result of rounding `null` be `NaN`. If `NaN` is being used as a sentinel for `null` here, we should be able to just copy the mask from the input column to the output column.

Reply:
Sure, we can make it explicit